invenio_oauthclient/views/client.py

```diff
@@ -39,28 +39,6 @@
 )
 
 
-@blueprint.record_once
```

Comment:
One thing I just realized, in this case this function using `@blueprint.record_once` would only run for the UI app and not the API app (i.e. the `/api` mount). Some of these cases are intentional (like here for example, we're setting template config variables). For other cases it might be though something that was overlooked or a even mistake.

This might hint that we need a UI and API variant of the "finalize_app" entrypoint? Similar to how we have `invenio_base.blueprints` and `invenio_base.api_blueprints`.

Reply:
should i have first a look, if there are already use-cases for `invenio_base.api_finalize_app`?

Reply:
I think it's good to have it and be conscious about its usage, so yes I mean it's meant to happen at some point that we need something running on the API app only.

Reply:
the [commit](https://github.com/inveniosoftware/invenio-app/blob/5db9f3349f4593a108d67fb1be07370f310ef419/invenio_app/factory.py#L91-L100) in invenio-app which introduce the `invenio_base.finalize_app` entrypoint is only applied to `create_ui` and `create_app` and not `create_api`. Should the `invenio_base.api_finalize_app` entrypoint then added to `create_api`? 

the `/api` endpoint is mentioned in `create_app`. does this mean that `invenio_base.api_finalize_app` should be added to `create_app` instead of `invenio_base.finalize_app`? As i read that code, that would be false, so it should remain as it is with `invenio_base.finalize_app`.